host/testcluster.go

```diff
@@ -203,10 +208,13 @@ func NewCluster(options *TestClusterConfig, logger log.Logger) (*TestCluster, er
 func setupShards(testBase persistencetests.TestBase, numHistoryShards int, logger log.Logger) {
 	// shard 0 is always created, we create additional shards if needed
 	for shardID := 1; shardID < numHistoryShards; shardID++ {
-		err := testBase.CreateShard(shardID, "", 0)
+		ctx, cancel := context.WithTimeout(context.Background(), defaultTestPersistenceTimeout)
```

Comment:
defer?

Reply:
It's a for loop, I can't use defer. `cancel()` is called on L214 and L217.